packages/stream_chat_flutter/lib/src/message_input.dart

```diff
@@ -447,101 +483,183 @@ class MessageInputState extends State<MessageInput> {
+                        ),
+                  ),
+                ],
+              ),
+            )
+          : null,
+      suffixIcon: Row(
+        children: [
+          if (_commandEnabled)
+            IconButton(
+              icon: StreamSvgIcon.closeSmall(),
+              splashRadius: 24,
+              padding: const EdgeInsets.all(0),
+              constraints: BoxConstraints.tightFor(
+                height: 24,
+                width: 24,
+              ),
+              onPressed: () {
+                setState(() => _commandEnabled = false);
+              },
+            ),
+          if (widget.sendButtonLocation == SendButtonLocation.inside)
+            _animateSendButton(context),
+        ],
+      ),
+    ).copyWith(
+      icon: passedDecoration?.icon,
+      labelText: passedDecoration?.labelText,
+      labelStyle: passedDecoration?.labelStyle,
+      helperText: passedDecoration?.helperText,
+      helperStyle: passedDecoration?.helperStyle,
+      helperMaxLines: passedDecoration?.helperMaxLines,
+      hintText: passedDecoration?.hintText,
+      hintStyle: passedDecoration?.hintStyle,
+      hintTextDirection: passedDecoration?.hintTextDirection,
+      hintMaxLines: passedDecoration?.hintMaxLines,
+      errorText: passedDecoration?.errorText,
+      errorStyle: passedDecoration?.errorStyle,
+      errorMaxLines: passedDecoration?.errorMaxLines,
+      floatingLabelBehavior: passedDecoration?.floatingLabelBehavior,
+      isCollapsed: passedDecoration?.isCollapsed,
+      isDense: passedDecoration?.isDense,
+      contentPadding: passedDecoration?.contentPadding,
+      prefixIcon: passedDecoration?.prefixIcon,
+      prefix: passedDecoration?.prefix,
+      prefixText: passedDecoration?.prefixText,
+      prefixIconConstraints: passedDecoration?.prefixIconConstraints,
+      prefixStyle: passedDecoration?.prefixStyle,
+      suffixIcon: passedDecoration?.suffixIcon,
+      suffix: passedDecoration?.suffix,
+      suffixText: passedDecoration?.suffixText,
+      suffixStyle: passedDecoration?.suffixStyle,
+      suffixIconConstraints: passedDecoration?.suffixIconConstraints,
+      counter: passedDecoration?.counter,
+      counterText: passedDecoration?.counterText,
+      counterStyle: passedDecoration?.counterStyle,
+      filled: passedDecoration?.filled,
+      fillColor: passedDecoration?.fillColor,
+      focusColor: passedDecoration?.focusColor,
+      hoverColor: passedDecoration?.hoverColor,
+      errorBorder: passedDecoration?.errorBorder,
+      focusedBorder: passedDecoration?.focusedBorder,
+      focusedErrorBorder: passedDecoration?.focusedErrorBorder,
+      disabledBorder: passedDecoration?.disabledBorder,
+      enabledBorder: passedDecoration?.enabledBorder,
+      border: passedDecoration?.border,
+      enabled: passedDecoration?.enabled,
+      semanticCounterText: passedDecoration?.semanticCounterText,
+      alignLabelWithHint: passedDecoration?.alignLabelWithHint,
+    );
```

Comment:
Consider using `merge` instead of `copyWith`.

Reply:
there's no InputDecoration.merge method
I can add an extension method

Reply:
done